mri/operators/fourier/utils.py

```diff
@@ -245,33 +245,50 @@ def generate_operators(data, wavelet_name, samples, mu=1e-06, nb_scales=4,
     return gradient_op, linear_op, prox_op, cost_op
 
 
-def get_stacks_fourier(kspace_loc):
+def get_stacks_fourier(kspace_loc, volume_shape):
     """Function that splits an incoming 3D stacked k-space samples
     into a 2D non-Cartesian plane and the vector containing the z k-space
-    values of all the plane and converts to stacks of 2D. This function also
-    checks for any issues of the incoming k-space pattern and if the stack
-    property is not satisfied.
+    values of the stacks acquiered and converts to stacks of 2D.
+    This function also checks for any issues of the incoming k-space
+    pattern and if the stack property is not satisfied.
     Stack Property: The k-space locations originate from a stack of 2D samples.
 
     Parameters
     ----------
-    ksapce_plane_loc: np.ndarray
-        the mask samples in the 3D Fourier domain.
-
+    kspace_loc: np.ndarray
+        Acquired 3D k-space locations : stacks of same non-Cartesian samples,
+        while Cartesian under-sampling on the stacks direction.
+    volume_shape: tuple
+        Reconstructed volume shape
     Returns
     ----------
-    ksapce_plane_loc: np.ndarray
+    kspace_plane_loc: np.ndarray
         A 2D array of samples which when stacked gives the 3D samples
     z_sample_loc: np.ndarray
         A 1D array of z-sample locations
     sort_pos: np.ndarray
         The sorting positions for opertor and inverse for incoming data
+    idx_mask_z: np.ndarray
+        contains the indices of the acquired Fourier planes (z direction)
     """
     # Sort the incoming data based on Z, Y then X coordinates
     # This is done for easier stacking
     sort_pos = np.lexsort(tuple(kspace_loc[:, i]
                                 for i in np.arange(3)))
     kspace_loc = kspace_loc[sort_pos]
+
+    # Find the mask used to sample stacks in z direction
+    full_stack_z_loc = \
```

Comment:
I think it's preferable to limit the use of `\`, you could do:
```python
full_stack_z_loc = convert_mask_to_locations(
  np.ones(volume_shape[2]),
)[:, 0]

Reply:
Ok, it will be on the next commit